### ISSUE TITLE
[1.3.3] Upgrade breaks app, 'require is not defined'

### ISSUE THREAD

**Issue by holmrenser** [meteor/meteor#7226]
After upgrading to 1.3.3 using ```meteor update``` my app instantly crashes.
Working on mac.
```
(STDERR)           
(STDERR) /Users/rensholmer/.meteor/packages/meteor-tool/.1.3.3.1ryifg6++os.osx.x86_64+web.browser+web.cordova/mt-os.osx.x86_64/dev_bundle/server-lib/node_modules/fibers/future.js:280
(STDERR) 						throw(ex);
(STDERR) 						      ^
(STDERR) ReferenceError: require is not defined
(STDERR)     at Package (packages/electronifiejs_pages.js:45:16)
(STDERR)     at packages/electronifiejs_pages.js:1296:4
(STDERR)     at packages/electronifiejs_pages.js:1303:3
(STDERR)     at /Users/rensholmer/Documents/Code/NODE/meteor/portal/.meteor/local/build/programs/server/boot.js:291:10
(STDERR)     at Array.forEach (native)
(STDERR)     at Function._.each._.forEach (/Users/rensholmer/.meteor/packages/meteor-tool/.1.3.3.1ryifg6++os.osx.x86_64+web.browser+web.cordova/mt-os.osx.x86_64/dev_bundle/server-lib/node_modules/underscore/underscore.js:79:11)
(STDERR)     at /Users/rensholmer/Documents/Code/NODE/meteor/portal/.meteor/local/build/programs/server/boot.js:133:5
=> Exited with code: 8
```

**Comment by benjamn** [comment]
What version were you updating from?

**Comment by holmrenser** [comment]
1.3.2

On 15 Jun 2016, at 19:00, Ben Newman <notifications@github.com<mailto:notifications@github.com>> wrote:


What version were you updating from?

-

**Comment by benjamn** [comment]
This appears to be a consequence of changes to the `coffeescript` package in Meteor 1.3.3 that allow raw JavaScript enclosed by back-ticks to be [compiled using `babel-compiler`](https://github.com/meteor/meteor/blob/9440636c1eb47feabd268744a552878f8cf40177/packages/coffeescript/plugin/compile-coffeescript.js#L77).

Unfortunately, sometimes Babel adds `require` calls to the code that it generates, in an attempt to fix certain runtime problems, or implement new ES2015 features. 

Since the `electronifiejs:pages` package contains [a file that contains back-ticks](https://github.com/electronifie/meteor-pages/blob/79b3a57e28091f2a43e1068b8c158d0a5f1d1ddf/lib/pages.coffee#L106), that file is passed through Babel, and ends up with `require("babel-runtime/helpers/typeof")` at the top.

But since `electronifiejs:pages` does not use the `modules` package, it doesn't have the `require` function, which leads to your error.

This is definitely a bug in the `coffeescript` package that should be fixed, but that will have to wait for the next `coffeescript` release.

In the meanwhile, I can think of two ways to fix this. Both approaches involve installing `electronifiejs:pages` locally, so do this first:

```sh
cd path/to/app
mkdir packages
cd packages
git clone https://github.com/electronifie/meteor-pages.git
```

Now that you have a local copy of `electronifiejs:pages`, you can either

1. (less recommended) make `electronifiejs:pages` use the `modules` package, by adding `api.use("modules")` to the `Package.onUse` section of `path/to/app/packages/meteor-pages/package.js`, or
2. (more recommended) remove the back-ticks from that line in [`pages.coffee`](https://github.com/electronifie/meteor-pages/blob/79b3a57e28091f2a43e1068b8c158d0a5f1d1ddf/lib/pages.coffee#L106):
```diff
diff --git a/lib/pages.coffee b/lib/pages.coffee
index 66337ee..af84f94 100755
--- a/lib/pages.coffee
+++ b/lib/pages.coffee
@@ -103,7 +103,7 @@
   
   constructor: (collection, settings = {}) ->
     unless @ instanceof Meteor.Pagination
-      throw new Meteor.Error "missing-new", "The Meteor.Pagination instance has to be initiated with `new`"
+      throw new Meteor.Error "missing-new", "The Meteor.Pagination instance has to be initiated with new"
     
     # Instance variables
```

**Comment by holmrenser** [comment]
Thanks for this workaround! This fixed the problem, however in the end I simply removed ```electronifiejs:pages```.

**Comment by benjamn** [comment]
This should be fixed in Meteor 1.3.3.1, which you can try by running `meteor update --release 1.3.3.1`. Feel free to reopen if the problem happens again.
